src/NHibernate/Linq/LinqExtensionMethodAttribute.cs

```diff
@@ -1,18 +1,34 @@
-using System;
-
 namespace NHibernate.Linq
 {
-	public class LinqExtensionMethodAttribute: Attribute
+	/// <summary>
+	/// Flag a method as being a SQL function call for the linq-to-nhibernate provider. Its
+	/// parameters will be used as the function call parameters. Unless <see cref="DBOnlyAttribute.DBOnly"/>
+	/// is explicitly setted to <see langword="true" />, the provider will execute the method when its does
+	/// arguments do not depend on the queried data.
+	/// </summary>
+	public class LinqExtensionMethodAttribute: DBOnlyAttribute
 	{
+		/// <summary>
+		/// Default constructor. The method call will be translated by the linq provider to
+		/// a function call having the same name than the method.
+		/// </summary>
 		public LinqExtensionMethodAttribute()
 		{
+			DBOnly = false;
```

Comment:
Having the method by default evaluated in memory whenever possible.

For performances reasons, it is usually better to evaluate in memory, once per request, all computation which can be, rather than letting the SQL server doing that. Things that are purely iterative computations are usually a job more suitable for iterative processing engines like .Net than for data querying engines.

Especially when it is a function call: for SQL servers, most functions are "black-boxes", and if they are not correctly flagged as deterministic, they will be evaluated for each row even if their actual result with same arguments will always be the same. Causing extension calls which were previously in-memory evaluated to always be evaluated in SQL as function could be a major breaking change for performances.

When writing Linq queries, it is frequent to write expressions that can be evaluated before hand. Changing the Linq provider for not evaluating them would cause anyone willing to optimize them to extract them in local variables before writing the query, which could be quite a hassle. That is the very purpose of the partial evaluating processor, evaluating all sub-expression which can be.